ckanext/xloader/tests/test_jobs.py

```diff
@@ -103,6 +103,66 @@ def test_xloader_data_into_datastore(self, cli, data):
         resource = helpers.call_action("resource_show", id=data["metadata"]["resource_id"])
         assert resource["datastore_contains_all_records_of_source_file"]
 
+    def test_download_resource_data_with_ckanext_xloader_site_url(self, cli, data):
+        # Set the ckanext.xloader.site_url in the config
+        with mock.patch.dict(toolkit.config, {'ckanext.xloader.site_url': 'http://xloader-site-url'}):
```

Comment:
This can be done more cleanly via `@pytest.mark.ckan_config` test annotations.

Reply:
I've not touched @kowh-ai 's tests. @ThrawnCA should I improve them?

Reply:
I think it's better to use the annotations than `with`.

I'm not really reviewing this in conjunction with the other PR, I'm just looking at it on its own merits.